zlib-rs/src/inflate.rs

```diff
@@ -1862,15 +1862,31 @@ fn inflate_fast_help_impl<const FEATURES: usize>(state: &mut State, _start: usiz
     let mut bad = None;
 
     if bit_reader.bits_in_buffer() < 10 {
-        bit_reader.refill();
+        if bit_reader.bytes_remaining() >= 8 {
```

Comment:
I added this check to make sure the safety precondition was met. But if there's any way to know for sure that `buffer.bytes_remaining()` is guaranteed to be >= 8 when this function is called, I can document that with a comment in place of the if-statement here.